src/components/postDropdown/container/postDropdownContainer.tsx

```diff
@@ -177,11 +182,137 @@ class PostDropdownContainer extends PureComponent {
           },
         });
         break;
+      case 'mute':
+        this._handleMuteUnmuteUser(!isMuted);
+        break;
+      case 'unmute':
+        this._handleMuteUnmuteUser(!isMuted);
+        break;
       default:
         break;
     }
   };
 
+  _handleMuteUnmuteUser = async (isMuteAction) => {
+    const { currentAccount, content } = this.props as any;
+    const username = content.author;
+    const isOwnProfile = !username || currentAccount.username === username;
+    console.log('isMuteAction,isOwnProfile : ', isMuteAction, isOwnProfile);
+
+    if(!isOwnProfile){
+      console.log('isMuteAction : ', isMuteAction);
+      
+      if (isMuteAction) {
+        this._muteUser();
+      } else {
+        this._handleFollowUnfollowUser();
+      }
+    }
+    
+  };
+  _muteUser = () => {
+    const { currentAccount, pinCode, dispatch, intl, content, onLoadPosts } = this.props as any;
+    const username = content.author;
+    const follower = currentAccount.name;
+    const following = username;
+   
+    ignoreUser(currentAccount, pinCode, {
+      follower,
+      following,
+    })
+      .then(() => {
+        const curMutes = currentAccount.mutes || [];
+        if (curMutes.indexOf(username) < 0) {
+          //check to avoid double entry corner case
+          currentAccount.mutes = [username, ...curMutes];
+        }
+        dispatch(updateCurrentAccount(currentAccount));
+        onLoadPosts(true); //fetch posts again after muting user
```

Comment:
instead of passing this prop from grad parent, perhaps better option would to hive content on feed, I think there is an option for that... the card can keep tracked of muted blocked users from currentAccount and hide content using that instead passing a prop deep inside.